web/third/bootstrap-typeahead/typeahead.js

```diff
@@ -499,6 +499,11 @@ import {get_string_diff} from "../../src/util";
       setTimeout(function () {
         if (!that.$container.is(':hover')) {
           that.hide();
+        } else {
+          // refocus the input if the user clicked on the typeahead
+          // so that clicking elsewhere registers as a blur and hides
+          // the typeahead.
+          that.$element.focus();
```

Comment:
Given the setTimeout, I worry a great deal that this will cause weird focus hopping behavior if the typeahead closed in the meantime.

Reply:
Oh right, thanks for catching that! 

I was able to reproduce such a scenario where with the mouse hovering the topic typeahead, if we shift tab focus to the stream input, the focus would then hop back to the topic input.

However, this (particular case) can be fixed by only refocusing if that typeahead is still shown... Have pushed this tweak!

(There still may be more edge cases where focus changes unnecessarily, but I haven't come across any yet)